modules/light-clients/08-wasm/keeper/msg_server.go

```diff
@@ -32,6 +33,22 @@ func (k Keeper) StoreCode(goCtx context.Context, msg *types.MsgStoreCode) (*type
 	}, nil
 }
 
+// RemoveCodeHash defines a rpc handler method for MsgRemoveCodeHash
+func (k Keeper) RemoveCodeHash(goCtx context.Context, msg *types.MsgRemoveCodeHash) (*types.MsgRemoveCodeHashResponse, error) {
+	if k.GetAuthority() != msg.Signer {
+		return nil, errorsmod.Wrapf(ibcerrors.ErrUnauthorized, "expected %s, got %s", k.GetAuthority(), msg.Signer)
+	}
+
+	found := types.HasCodeHash(goCtx, msg.CodeHash)
+
+	err := ibcwasm.CodeHashes.Remove(goCtx, msg.CodeHash)
```

Comment:
This doesn't remove the code from the vm cache, should we also be calling https://github.com/CosmWasm/wasmvm/blob/main/lib.go#L69

Reply:
I don't think we should remove it. Wouldn't this break already existing light clients using the removed code?

Reply:
Yeah, I think you're right it would, good catch! As you could no longer instantiate the contract to call into right? You'd be kind of in no man's land. Maybe that could also be a good thing if this was being used as some sort of killswitch, but once its removed I don't see much of a path for recovery - probably upgrade the channel to a new connection to recover funds would be the only option. 

Happy to keep it as you have 👍